### DISCUSSION TITLE
Remove id column of database column mt_metadata

### DISCUSSION THREAD

**Discussion by matthiasstraka** [gerbera/gerbera#1875] [Ideas]
I analyzed the `mt_metadata` database table and in my opinion the `id` column has no function anymore. It is an auto-incremented primary key that is not queried in any select nor is it inserted or modified (look for `METADATA_TABLE` in the code). Also, this table happens to be the one with the largest storage size for me. I propose to do the following:

- Drop the column id
- Change the primary key to (item_id, property_name)
- Drop the metadata_item_id index (the new primary key will be sufficient as it indexes by item_id)
- Upgrade scripts can be implemented without the need to for data transformation (just key/index operations)
- The storage-requirement will drop significantly (my MariaDB reports 9.2MB data and 4.2 MB index size for ~200000 rows -> we can remove the index and save 200K * 4 = 800kB of int-values)

Please let me know if this modification appears useful to you and I can prepare a PR. I just don't want start doing it without knowing it will become merged eventually.

**Comment by KarlStraussberger** [comment]
Actually, I've been looking into the multi-valued properties topic (two artists, composers etc.). In order to do it correctly, there is need for a separate key.

**Reply by matthiasstraka** [reply]
Thanks for the reply, I will leave the table untouched then.
